### PR TITLE
IBP-4059 recategorize obs unit id as experimental design variable

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review

Included:
https://github.com/IntegratedBreedingPlatform/Fieldbook/pull/1132
https://github.com/IntegratedBreedingPlatform/Commons/pull/367
https://github.com/IntegratedBreedingPlatform/BMSAPI/pull/770

### CODE REVIEW

File: src/main/resources/liquibase/crop_changelog/22_3_0.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+	<changeSet author="cuenyad" id="v22.3.0-1">
+		<preConditions onFail="MARK_RAN">
+				<sqlCheck expectedResult="1">
+					SELECT count(1) > 0 FROM CVTERMPROP WHERE CVTERM_ID = 8201 AND value = 'Germplasm Descriptor';
+				</sqlCheck>
+		</preConditions>
+		<comment>Recategorize OBS_UNIT_ID as Experimental Design variable</comment>
+		<sql dbms="mysql" splitStatements="true">
+			set @OBS_UNIT_ID = (select cvterm_id from cvterm where name ='OBS_UNIT_ID' and cv_id = 1040 and is_obsolete = 0);
```
Comment (clarysabel) [inline src/main/resources/liquibase/crop_changelog/22_3_0.xml:15]: @cuenyad we dont need to check by is_obsolete now as it was removed from the unique key

Comment (cuenyad) [reply src/main/resources/liquibase/crop_changelog/22_3_0.xml]: done!.

File: src/main/java/org/generationcp/middleware/api/brapi/v2/observationunit/ObservationUnitServiceImpl.java [hunk 2]
```diff
@@ -317,6 +321,21 @@ public List<String> importObservationUnits(final String crop, final List<Observa
 		return observationUnitDbIds;
 	}
 
+	private void checkIfExistsObsUniIdVariable(final Map<Integer, DmsProject> plotDatasetMap) {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/brapi/v2/observationunit/ObservationUnitServiceImpl.java:324]: @cuenyad the name is not clear enough, it is not only checking right? let's find a name

File: src/main/resources/liquibase/crop_changelog/22_3_0.xml [hunk 3]
```diff
@@ -62,4 +62,40 @@
...
+		</preConditions>
+		<comment>Recategorize OBS_UNIT_ID as Experimental Design variable</comment>
+		<sql dbms="mysql" splitStatements="true">
+			set @OBS_UNIT_ID = (select cvterm_id from cvterm where name ='OBS_UNIT_ID' and cv_id = 1040);
+
+			# update the type of variable as Germplasm Descriptor.
+								  UPDATE CVTERMPROP SET value = 'Experimental Design' WHERE CVTERM_ID = @OBS_UNIT_ID;
+
+			# update the Germplasm Descriptor cvterm_id by Experimental Design in the projectprop table.
+				  UPDATE projectprop SET type_id = 1810  WHERE variable_id = @OBS_UNIT_ID;
```
Comment (clarysabel) [inline src/main/resources/liquibase/crop_changelog/22_3_0.xml:79]: @cuenyad for safety I think we should add the type_id condition in the where clause

File: src/main/resources/liquibase/crop_changelog/22_3_0.xml [hunk 4]
```diff
@@ -62,4 +62,40 @@
...
+				<sqlCheck expectedResult="1">
+					SELECT count(1) > 0 from projectprop where variable_id = (select cvterm_id from cvterm where name ='OBS_UNIT_ID' and cv_id = 1040) and
+							project_id in (SELECT project_id FROM project where program_uuid is null);
+				</sqlCheck>
+		</preConditions>
+		<comment>Remove OBS_UNIT_ID variable from the study templates</comment>
+		<sql dbms="mysql" splitStatements="true">
+			set @OBS_UNIT_ID = (select cvterm_id from cvterm where name ='OBS_UNIT_ID' and cv_id = 1040);
+
+			delete from projectprop where variable_id = @OBS_UNIT_ID and
```
Comment (clarysabel) [inline src/main/resources/liquibase/crop_changelog/22_3_0.xml:95]: @cuenyad same here, just to make sure we remove for the type_id Germplasm Descriptor, also in the precondition

**Pull request merged**